### PR TITLE
Emphasize still talking about sending

### LINKED ISSUE #3933: Recommended behavior for reserved streams
In Section 6.2.3, we recommend the following about the reserved unidirectional stream types:
> When terminating abruptly, the error code H3_NO_ERROR or a reserved error code (Section 8.1) SHOULD be used.

However, this has a fatal flaw -- it recommends special handling for the grease types, which is *precisely* the behavior we're trying to disincent with grease.

Instead, I think we should move the recommendation to use reserved error codes to the guidance on what to do with unknown stream types, which is currently:
> Recipients of unknown stream types MAY abort reading of the stream with an error code of H3_STREAM_CREATION_ERROR, but MUST NOT consider such streams to be a connection error of any kind.

This has the same net effect on the grease streams.  The change in behavior is on actual unknown-but-real stream types, which will now get grease error codes in response.

### PR DESCRIPTION
**Pull request by MikeBishop**
I totally misread this passage, so let's make it more emphatic which side of the stream we're talking about.  Fixes #3933.

### CODE REVIEW

File: draft-ietf-quic-http.md [hunk 1]
```diff
@@ -1165,10 +1165,11 @@ desired. They MAY also be sent on connections where no data is currently being
...
 receipt.
 
-The payload and length of the stream are selected in any manner the
-implementation chooses.  Implementations MAY terminate these streams cleanly, or
-MAY abruptly terminate them.  When terminating abruptly, the error code
-H3_NO_ERROR or a reserved error code ({{http-error-codes}}) SHOULD be used.
+The payload and length of the stream are selected in any manner the sending
+implementation chooses.  When sending a reserved stream type, the implementation
+MAY either terminate the stream cleanly or reset it.  When resetting the stream,
+the error code H3_NO_ERROR or a reserved error code ({{http-error-codes}})
```
Comment (martinthomson) [inline draft-ietf-quic-http.md:1171]:
```suggestion
either H3_NO_ERROR or a reserved error code ({{http-error-codes}})
```

File: draft-ietf-quic-http.md [hunk 2]
```diff
@@ -1165,10 +1165,11 @@ desired. They MAY also be sent on connections where no data is currently being
 transferred. Endpoints MUST NOT consider these streams to have any meaning upon
 receipt.
 
-The payload and length of the stream are selected in any manner the
-implementation chooses.  Implementations MAY terminate these streams cleanly, or
-MAY abruptly terminate them.  When terminating abruptly, the error code
-H3_NO_ERROR or a reserved error code ({{http-error-codes}}) SHOULD be used.
+The payload and length of the stream are selected in any manner the sending
+implementation chooses.  When sending a reserved stream type, the implementation
+MAY either terminate the stream cleanly or reset it.  When resetting the stream,
```
Comment (martinthomson) [inline draft-ietf-quic-http.md:1170]:
```suggestion
MAY either allow the stream to terminate cleanly or reset it.  When resetting the stream,
```

Comment (MikeBishop) [reply draft-ietf-quic-http.md:1170]: On the sender side, though, either is an explicit action; it's not something you just allow to happen.

Comment (janaiyengar) [reply draft-ietf-quic-http.md:1170]: @martinthomson : Is your point that terminating cleanly is application action where resetting is transport action?

Comment (martinthomson) [reply draft-ietf-quic-http.md:1170]: As a receiver you have to let the stream terminate cleanly, the only other action available is a stream reset.

Presumably the sender knows what they are doing, so I wanted to concentrate on the reaction the receiver might have.

Comment (MikeBishop) [reply draft-ietf-quic-http.md:1170]: The generic advice we give for unknown stream types (which this should be) is to reply with a STOP_SENDING.  On the receiver side, you shouldn't be identifying these as grease streams.

Comment (janaiyengar) [reply draft-ietf-quic-http.md:1170]: This is entirely about what the sender does, right?

**Pull request merged**